PoC/runtimes/kafka/kafka-consumer/src/main/java/org/eclipse/tractusx/edc/kafka/consumer/EdrTokenCallbackHandler.java

```diff
@@ -0,0 +1,177 @@
+ */
+@Slf4j
+public class EdrTokenCallbackHandler implements AuthenticateCallbackHandler {
+
+
+    private boolean isInitialized = false;
+    private String scopeClaimName;
+    private final EdrDataProvider edrDataProvider;
+
+    /**
+     * Constructor with default EdrDataProvider implementation
+     */
+    public EdrTokenCallbackHandler() {
+        this.edrDataProvider = new DefaultEdrDataProvider();
+    }
+
+    /**
+     * Constructor with custom EdrDataProvider for testing
+     *
+     * @param edrDataProvider custom provider implementation
+     */
+    public EdrTokenCallbackHandler(final EdrDataProvider edrDataProvider) {
+        this.edrDataProvider = edrDataProvider;
+    }
+
+    @Override
+    public void configure(final Map<String, ?> configs, final String mechanism, final List<AppConfigurationEntry> jaasConfigEntries) {
+        ConfigurationUtils cu = new ConfigurationUtils(configs, mechanism);
+        scopeClaimName = cu.get(SASL_OAUTHBEARER_SCOPE_CLAIM_NAME);
+        isInitialized = true;
+    }
+
+    @Override
+    public void handle(final Callback[] callbacks) throws UnsupportedCallbackException, IOException {
+        checkInitialized();
+
+        for (Callback callback : callbacks) {
+            if (callback instanceof OAuthBearerTokenCallback tokenCallback) {
+                try {
+                    handleTokenCallback(tokenCallback);
+                } catch (InterruptedException e) {
+                    log.warn("Interrupted while handling OAuthBearerTokenCallback", e);
+                    Thread.currentThread().interrupt();
+                }
+            } else {
+                throw new UnsupportedCallbackException(callback, "Unsupported callback type");
+            }
+        }
+    }
+
+    private void handleTokenCallback(final OAuthBearerTokenCallback callback) throws IOException, InterruptedException {
+        log.debug("Handling OAuthBearerTokenCallback");
+        String accessToken = fetchAccessTokenFromEDC();
+        OAuthBearerToken oAuthBearerToken = toOAuthBearerToken(accessToken);
+        callback.token(oAuthBearerToken);
+        log.debug("OAuth bearer token successfully provided to callback");
+    }
+
+    private OAuthBearerToken toOAuthBearerToken(final String accessToken) {
+        log.debug("Converting JWT access token to OAuthBearerToken");
+        try {
+            DecodedJWT jwt = JWT.decode(accessToken);
+            long issuedAt = jwt.getIssuedAtAsInstant().toEpochMilli();
+            long expiresAt = jwt.getExpiresAtAsInstant().toEpochMilli();
+            String subject = jwt.getSubject();
+            String scopeClaim = jwt.getClaim(scopeClaimName).asString();
+            Set<String> scopes = Set.of(scopeClaim);
+
+            return new BasicOAuthBearerToken(accessToken, scopes, expiresAt, subject, issuedAt);
+        } catch (Exception e) {
+            log.error("Failed to decode JWT token", e);
```

Comment:
In case of generic Exception handling "Unexpected error during JWT decoding" should be used instead because the error cause NPE, decode exception could not be determinded